Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/model/builder/impl/PatientSearchResultsModelBuilderImpl.java

```diff
@@ -110,28 +114,26 @@ private void extractAndAddPatient(PRPAMT201310UV02Patient msgPatient) {
     private void extractNames(PRPAMT201310UV02Person person, Patient patient) {
         if (person.getName() != null && person.getName().size() > 0 && person.getName().get(0) != null
                 && person.getName().get(0).getContent() != null && person.getName().get(0).getContent().size() > 0) {
-
-            boolean firstNameFound = false;
-            boolean secondNameFound = false;
-
             for (Serializable object : person.getName().get(0).getContent()) {
                 if (object instanceof JAXBElement<?>) {
                     Object nameValue = ((JAXBElement<?>) object).getValue();
 
                     if (nameValue instanceof EnExplicitFamily) {
                         patient.setLastName(((EnExplicitFamily) nameValue).getContent());
                     } else if (nameValue instanceof EnExplicitGiven) {
-                        if (!firstNameFound) {
-                            patient.setFirstName(((EnExplicitGiven) nameValue).getContent());
-                            firstNameFound = true;
-                        } else if (!secondNameFound) {
-                            patient.setMiddleName(((EnExplicitGiven) nameValue).getContent());
-                            secondNameFound = true;
+                        //the first given name is first name, following by middle name
+                        String givenName = ((EnExplicitGiven) nameValue).getContent();
+                        if (StringUtils.hasText(patient.getFirstName())){
+                            patient.setMiddleName(givenName);
```

Comment:
We do need middle name.  However, if the given name list has two entry, the first one is first name and second one is middle name